test/bntest.c

```diff
@@ -2219,6 +2219,36 @@ static int test_mpi(int i)
     return st;
 }
 
+static int test_bin2zero(void)
+{
+    unsigned char input[] = { '\0' };
```

Comment:
`unsigned char input = 0;` and use `&input` later?

Regardless, `'\0'` is definitely wrong, it's a `char`.  Just a plain simple `0` is best.

Reply:
Okie